### PR TITLE
Support completion for operator in #case #is sections

### LINKED ISSUE #715: Completion operator for #case #is
Given this Qute template:

```
{#switch name}
	{#case | "abcd"}
{/switch}
```

The completion triggered at `|` should see available operators. See available operators at https://github.com/quarkusio/quarkus/blob/9366bc89ad69ffd4ebef003b31b0e7cff46d8689/independent-projects/qute/core/src/main/java/io/quarkus/qute/WhenSectionHelper.java#L174

### PR DESCRIPTION
**Pull request by JessicaJHee**
Fixes #715 

Signed-off-by: Jessica He <jhe@redhat.com>

### CODE REVIEW

**Comment by JessicaJHee** [comment]
I'm not sure if we want to complete with all the possible operators (including the alias). I have created a list of operators that make the most sense to me in CaseSection (completionOperators) for now. Please let me know of other suggestions.

File: qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/completions/QuteCompletionInExpressionWithWhenSectionTest.java [hunk 1]
```diff
@@ -111,8 +133,7 @@ public void propertyPartWhenWithNotInNewLine() throws Exception {
 				"{/when}";
 		testCompletionFor(template, 3, //
 				c("OFF : MachineStatus", "OFF", r(3, 10, 3, 10)), //
-				c("BROKEN : MachineStatus", "BROKEN", r(3, 10, 3, 10)),
-				c("in : MachineStatus", "in", r(3, 10, 3, 10)));
+				c("BROKEN : MachineStatus", "BROKEN", r(3, 10, 3, 10)), c("in : MachineStatus", "in", r(3, 10, 3, 10)));
```
Comment (angelozerr) [inline qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/completions/QuteCompletionInExpressionWithWhenSectionTest.java:136]: Please use // at the end of the line to keep formatting

**Comment by angelozerr** [comment]
> I'm not sure if we want to complete with all the possible operators (including the alias). 

@mkouba what do you think about thatwhen you have 

`{#case |}` and user open completion on `|`, the @JessicaJHee PR shows allowed parameter without alias. What is the best completion to do : 

 * show all operators with aliases?
 * show operators without aliases?

**Comment by mkouba** [comment]
> * show all operators with aliases?
> * show operators without aliases?

I'm not sure either but I tend to support the idea of showing all operators...

**Comment by angelozerr** [comment]
> I'm not sure either but I tend to support the idea of showing all operators...

Ok thanks for your feedback,let's show all operators in the completion.

**Comment by angelozerr** [comment]
@JessicaJHee please see my POC at https://github.com/redhat-developer/quarkus-ls/pull/737

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/parser/template/sections/CaseSection.java [hunk 2]
```diff
@@ -74,6 +85,15 @@ public CaseSection(int start, int end) {
 		super(tag, start, end);
 	}
 
+	@Override
+	protected void initializeParameters(List<Parameter> parameters) {
+		// All parameters can have expression (ex : {#if age=10} except operators
```
Comment (angelozerr) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/parser/template/sections/CaseSection.java:90]: // All parameters can have expression (ex : {#case OFF})

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCompletions.java [hunk 3]
```diff
@@ -173,6 +173,24 @@ public CompletableFuture<CompletionList> doComplete(Template template, Position
...
+	 * Returns true if completion is allowed at the current offset and section.
+	 *
+	 * @param parameter the parameter.
+	 * @param offset    the offset.
+	 *
+	 * @return true if completion is allowed at the current offset and section.
+	 */
+	public boolean isCompletionAllowed(Parameter parameter, int offset) {
+		if (Section.isCaseSection(parameter.getOwnerSection())) {
+			return true;
```
Comment (angelozerr) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCompletions.java:186]: // {#case O|FF}

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCompletions.java [hunk 4]
```diff
@@ -173,6 +173,24 @@ public CompletableFuture<CompletionList> doComplete(Template template, Position
...
+		}
+		if (parameter.isAfterAssign(offset)) {
+			return true;
```
Comment (angelozerr) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCompletions.java:189]: // {#let name=va|lue}

**Comment by angelozerr** [comment]
It works like a charm, great job @JessicaJHee !

**Pull request merged**